bot/src/main.rs

```diff
@@ -1528,48 +1528,54 @@ fn classify(
                 reporting_counts.insert(*pk, HashMap::from([(epoch - 1, *passed)]));
             });
             number_sampled_epochs = 1;
+        } else {
+            notes.push("Could not get reported performance metrics".to_string());
         };
 
         let mut number_loops = 0;
-
-        while number_sampled_epochs < NUM_SAMPLED_REPORTING_EPOCHS as u64 {
+        let max_epochs_to_check = 20;
+        while number_sampled_epochs < NUM_SAMPLED_REPORTING_EPOCHS as u64
+            && number_loops < max_epochs_to_check
+        {
             let reporting_epoch = epoch - 2 - number_loops;
 
             // Fetch from wiki repo
             if let Some(epoch_classification) = EpochClassification::load_if_validators_classified(
                 reporting_epoch + 1,
                 &config.cluster_db_path(),
             )? {
+                // Whether any "passed" records are found. If none are found, don't use the epoch.
```

Comment:
where does the final designation of whether this epoch will be used or not happen?

Reply:
https://github.com/solana-foundation/stake-o-matic/blob/acc181ed4348df97c84794ef94007023e0869b06/bot/src/main.rs#L1568-L1574

I'll rename this variable and add some comments to make this more clear.